### ISSUE TITLE
Accessibility issue - current date should use aria-current="date"

### ISSUE THREAD

**Issue by SKaplan01** [Hacker0x01/react-datepicker#3229]
**Describe the bug**

When a screenreader user navigates through the calendar, the current date is not announced as the current date.

**To Reproduce**

1. Go to the react-datepicker demo (https://reactdatepicker.com/)
2. Turn on a screenreader (I used Voiceover)
3. Open the calendar and press tab, focus will land on current date and screenreader will read something like "Choose Wednesday September 29, 2021, button"

**Expected behavior**

When the focus is on the current date, the screen reader should read something like "Choose Wednesday September 29, 2021, current date, button"

aria-current="date" can be added to the current date button.
